packages/component-library/src/components/plmg-button/plmg-button.tsx

```diff
@@ -159,6 +164,44 @@ export class Button {
     if (newValue && typeof newValue !== 'string')
       throw new Error('target must be a string');
   }
+
+  /**
+   * Define button's left icon.
+   *
+   * When providing an icon name to this prop, the corresponding icon will be displayed.
+   * it will be placed to the left of the text slot.
+   */
+  @Prop() iconLeft: string | undefined = undefined;
+
```

Comment:
Shouldn't we have some @Watch things for all these new props? Like making sure the string provided is a valid icon, and making sure a label is provided for the iconCenter etc.

Reply:
You're right, we could check the given icon. But I'm afraid that will lead to bigger package size, if we include the exhaustive list of supported icons :confused: Do you think the benefit of choosing from such a list is worth it ?

Reply:
I've added @Watch to check the type (string).

Reply:
Yeah I think that's probably reasonable. Of course it would be lovely for intellisense to tell me which icons I can use whenever I use the component, but you're right, it may not be worth it. I APPROVE!

Reply:
![image](https://user-images.githubusercontent.com/55230735/161507715-37954317-8352-48ef-86fa-24f0f463066a.png)

It's 64,7kB :thinking: Maybe we should include it.
Since it's only typing definitions, it's not going to increase the size of the product using the package, right ? :)

Reply:
Well, after some investigation: 
- the type declaration is already included in the build: it is in `packages/component-library/dist/types/components/plmg-svg-icon/icon.enum.d.ts`
- The main type file is `packages/component-library/dist/types/components.d.ts`
- The main type file imports `ICON` but does not re-export it

Therefore, the enumeration is not accessible :confused:

Reply:
I believe our issue is similar to https://github.com/ionic-team/stencil/issues/3258
I tried exporting the enum differently, adding the re-export in `src/index.ts` but I didn't manage to use it in a react example app :confused: